src/test/scala/chiselTests/ConnectSpec.scala

```diff
@@ -32,6 +32,7 @@ class PipeInternalWires extends Module {
 
 class CrossConnectTester(inType: Data, outType: Data) extends BasicTester {
   val dut = Module(new CrossConnects(inType, outType))
+  dut.io := DontCare
```

Comment:
This seems odd: CrossConnects.io is bidirectional, I'm surprised this doesn't error (or does it?) out when assigning DontCare to it.

Reply:
Yeah it works fine. Awkwardly just connecting to the actual input port errors because it's a Clock and apparently connecting DontCare to Clocks directly is an error

Reply:
Wait what is this code doing then? DontCare shouldn't have any effect on outputs, and the bundle connect should recurse to the inputs. Any thoughts why the behavior differs? That also sounds like a bug...

Reply:
It does recurse and mark both as invalid including the output. Firrtl allows this, it's just ignored.

I think it is sensible to mark bidirectional aggregates DontCare. IMO it's a useful shorthand rather than having to call out each submodule input or your own module output. We should fix the mismatch between marking a clock DontCare yet still being able to just mark it's parent aggregate DontCare.

Let's discuss these issues before 3.1!